src/main/java/name/neuhalfen/projects/crypto/bouncycastle/openpgp/keys/callbacks/Rfc4880KeySelectionStrategy.java

```diff
@@ -116,52 +119,61 @@ public PGPPublicKey selectPublicKey(PURPOSE purpose, String uid, KeyringConfig k
 
     final PGPSecretKeyRingCollection secretKeyRings = keyringConfig.getSecretKeyRings();
 
+    PGPPublicKey publicKey = null;
```

Comment:
Add a comment to the method (if there is none yet) to say that the _last_ matching key is selected ("use the newest").